satpy/readers/fci_l1c_nc.py

```diff
@@ -686,20 +697,19 @@ def calibrate_rad_to_refl(self, radiance, key):
                 "channel effective solar irradiance set to fill value, "
                 "cannot produce reflectance for {:s}.".format(measured))
             return radiance * np.float32(np.nan)
-
-        sun_earth_distance = np.mean(
-            self.get_and_cache_npxr("state/celestial/earth_sun_distance")) / 149597870.7  # [AU]
-
-        # TODO remove this check when old versions of IDPF test data (<v5) are deprecated.
-        if sun_earth_distance < 0.9 or sun_earth_distance > 1.1:
-            logger.info("The variable state/celestial/earth_sun_distance contains unexpected values"
-                        "(mean value is {} AU). Defaulting to 1 AU for reflectance calculation."
-                        "".format(sun_earth_distance))
-            sun_earth_distance = 1
-
+        sun_earth_distance = self._compute_sun_earth_distance()
         res = 100 * radiance * np.float32(np.pi) * np.float32(sun_earth_distance) ** np.float32(2) / cesi
         return res
 
+    def _compute_sun_earth_distance(self) -> float:
```

Comment:
this could be cached_property instead (see decorator of `orbital_param` as example)